methods/articles.coffee

```diff
@@ -2,28 +2,31 @@ Articles = require '/imports/collections/articles.coffee'
 { regexEscape } = require '/imports/utils'
 
 Meteor.methods
-  addEventSource: (source) -> #eventId, url, publishDate, publishDateTZ
+  addEventSource: (source, eventId) -> #eventId, url, publishDate, publishDateTZ
     user = Meteor.user()
     if user and Roles.userIsInRole(user._id, ['admin'])
       # Check if Document is in collection
       sourceQuery = url: $regex: "#{regexEscape(source.url)}$"
       existingSource = Articles.findOne(sourceQuery)
       if existingSource
+        #if this source is already in the DB and the userEvent isn't already associated - push it
         Articles.update sourceQuery,
-          $set: userEventId: source.userEventId
+          $addToSet: userEventIds: eventId
+        Meteor.call("editUserEventArticleCount", eventId, 1)
+        existingSource._id
       else
         if source.url
           insertArticle =
             url: source.url
             title: source.title
-            userEventId: source.userEventId
+            userEventIds: [eventId]
           insertArticle = source
```

Comment:
@nathanathan - that section of code only runs when you are inserting a new article so there shouldn't be a `userEventIds` property to wipe our, correct?  There is an if statement above that that handles updating `userEventIds` on existing sources so that that field isn't wiped out.

Are you still seeing the bug you mentioned earlier?